src/networking.c

```diff
@@ -399,6 +399,25 @@ void addReplyErrorLength(client *c, const char *s, size_t len) {
 
 /* Do some actions after an error reply was sent (Log if needed, updates stats, etc.) */
 void afterErrorReply(client *c, const char *s, size_t len) {
+    struct redisError *err = NULL;
+    /* Errors that occurred during command processing */
+    if (c->cmd){
+        c->cmd->failed_calls++; /* Update per command error stats */
+        err = lookupError(shared.commanderror);
+        if (err) err->count++;
+    }
+    /* Authorization error counter (commands blocked by ACLs) */
+    if (!strncmp(s, "-NOPERM", 7)){
+        err = lookupError(shared.authorizationerror);
+        if (err) err->count++;
+    /* Authentication errors */
+    } else if (!strncmp(s, "AUTH", 4) ||
+               !strncmp(s, "-NOAUTH", 7) ||
+               !strncmp(s, "-WRONGPASS", 10)){
```

Comment:
why isn't `AUTH` (the first error) prefixed with `-`?
```suggestion
    } else if (!strncmp(s, "AUTH", 4) ||
               !strncmp(s, "-NOAUTH", 7) ||
               !strncmp(s, "-WRONGPASS", 10))
    {
```

Reply:
Can we also just move the pointer one character in, we know the first character must be "-"

Reply:
Also, isn't this duplicative of the ACL log?

Reply:
> why isn't `AUTH` (the first error) prefixed with `-`?

@oranagra due to "AUTH" error string not starting with `-`. The `-` is added afterwards but non included in the error string, within addReplyErrorLength() line `if (!len || s[0] != '-') addReplyProto(c,"-ERR ",5);` .
Noneteless if we follow the new approach this is accounted for...

Reply:
ok, this means that the text "AUTH" which we matched against, is not the error code, it's just a part of the error text which happens to be uppercase.
i think it means we need to move `afterErrorReply` to be called in the different place, since it takes an error message, but it's no the final error message.